lib/models/user_data.dart

```diff
@@ -28,3 +29,38 @@ abstract class UserData extends ChangeNotifier {
     ''';
   }
 }
+
+/// Extension [TimeCheck] on [UserData].
+///
+/// This extension provides additional methods related to time checking for [UserData] objects.
+extension TimeCheck on UserData {
+  /// Checks if the current time is within the pickup range for a [Canteen].
+  ///
+  /// This method is only applicable if the [UserData] instance is a [Canteen].
+  /// It splits the pickup times into parts, creates [DateTime] objects for them,
+  /// and checks if the current time is within this range.
+  ///
+  /// Returns `true` if the current time is within the pickup range, `false` otherwise.
+  bool isCurrentTimeWithinPickupRange() {
+    if (this is Canteen) {
+      Canteen canteen = this as Canteen;
+      DateTime now = DateTime.now();
+      DateTime timeFrom = _getTime(canteen.pickUpFrom, now);
+      DateTime timeWithin = _getTime(canteen.pickUpWithin, now);
+      return now.isAfter(timeFrom) && now.isBefore(timeWithin);
+    }
+    return false;
+  }
+
+  /// Helper method to create a [DateTime] object from a time string.
+  ///
+  /// The time string should be in the format 'HH:mm'. The [DateTime] object
+  /// is created with the current date and the time from the time string.
+  ///
+  /// Returns a [DateTime] object with the current date and the time from the time string.
+  DateTime _getTime(String time, DateTime now) {
```

Comment:
I would either separate it to another file and move it to helpers directory, or make it fileprivate (is this possible in dart?)

Reply:
Yes, I moved it to `DateTimeUtils` and made it static and public. In Dart, a method marked with an underscore (_) is considered private. Is that ok?